Feat/firmware switch wipe

komret opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->
It does not look good, but modals will be redesigned soon and this should do until then.
I used red instead of yellow for now because we don't have a yellow button variant - this is up for discussion.
Perhaps the text could be adjusted a bit in case the device is not backed up.

## Related Issue

Resolve #9679 

## Screenshots:

![Screenshot 2023-10-30 at 18 33 12](https://github.com/trezor/trezor-suite/assets/42465546/e3914cd2-fa66-4596-8f5b-12a4c3a42d9e)
![Screenshot 2023-10-30 at 18 33 24](https://github.com/trezor/trezor-suite/assets/42465546/99170798-1094-4620-8d67-a6420d261487)

komret commented:
There seems to be a bug, when firmware with a different vendor header is installed, the modal gets stuck on "Restarting Trezor". I am testing it with a T1 custom firmware. I am investigating it now.

EDIT: It happens because of [this condition](https://github.com/trezor/trezor-suite/commit/e157076273d65f46220b60333d4366bd54c15df0). Device ID resets when the device is wiped.

matejkriz commented on packages/suite/src/views/firmware/index.tsx:
```diff
@@ -50,6 +51,8 @@ export const Firmware = ({ shouldSwitchFirmwareType }: FirmwareProps) => {
     const dispatch = useDispatch();
 
     const deviceModelInternal = device?.features?.internal_model;
+    const deviceWillBeWiped =
```
nit: I guess it deserves a comment here - T2B1 get wiped switching between Universal and Bitcoin-only because different FW vendor headers as security feature

komret replied:
e1268df4cf59d4a37b84b8a19e4f00d00d13254a

komret commented:
@matejkriz in e1268df4cf59d4a37b84b8a19e4f00d00d13254a I changed button colour from red to green to make it look less scary and added some placeholder translation keys for a scenario when the device will be wiped and is not backed up. These changes have been discussed with @Hermez-cz The icons should be replaced with yellow ones during redesign.

komret commented:
/rebase

matejkriz commented:
approving again latest changes, but I haven't tested them, just checked the code.
